prince-of-versions/src/main/java/co/infinum/princeofversions/JsonConfigurationParser.java

```diff
@@ -125,8 +124,13 @@ private void parseToBuilder(JSONObject data, PrinceOfVersionsConfig.Builder buil
                         return; //return after finding the first feasible update
                     }
                 }
+                if (android.length() > 0) {
+                    throw new RequirementsNotSatisfiedException("No feasible updates in JSON!");
```

Comment:
Nice catch, however could we throw some other exception? We should use `RequirementsNotSatisfiedException` only in the case when requirements aren't satisfied.
We could use IllegalArgumentException, but please do use a descriptive message that will be clear to user that there is no any objects in the array.

Reply:
I do agree, but I used `RequirementsNotSatisifiedException` because this if will only be invoked if there is no feasible updates in array. If there is any feasible update in JSON array, the return function would be called and this part of code wouldn't be invoked. Did you thought something else? The update is not feasible when the requirements are not satisfied.

Reply:
My bad, I thought exception is thrown if android array is empty. Can we do that if array is empty?

Reply:
Sure 👍